File: packages/python/examples/pytest/calculator_test.py
from os import getenv

from pytest import fixture

alumnium_driver = getenv("ALUMNIUM_DRIVER", "selenium")


@fixture(autouse=True)
def learn(al):
    # Mistral skips '+' button.
    al.learn(
        goal="4 / 2 =",
        actions=[
            "click button '4'",
            "click button '÷'",
            "click button '2'",
            "click button '='",
        ],
    )
    yield
    al.clear_learn_examples()


def test_addition(al, navigate):
    navigate("https://seleniumbase.io/apps/calculator")
    al.do("2 + 2 =")
    assert al.get("calculator result from textfield") == 4


def test_subtraction(al, navigate):
    navigate("https://seleniumbase.io/apps/calculator")
    al.do("5 - 3 =")
    assert al.get("calculator result from textfield") == 2


def test_multiplication(al, navigate):
    navigate("https://seleniumbase.io/apps/calculator")
    al.do("3 * 4 =")
    assert al.get("calculator result from textfield") == 12


def test_division(al, navigate):
    navigate("https://seleniumbase.io/apps/calculator")
    al.do("8 / 2 =")
    assert al.get("calculator result from textfield") == 4

File: packages/java/src/test/java/ai/alumnium/system/CalculatorTest.java
package ai.alumnium.system;

import ai.alumnium.Provider;
import java.util.List;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeAll;
import org.junit.jupiter.api.Test;

public class CalculatorTest extends BaseTest {

  private static final String CALCULATOR_URL = "https://seleniumbase.io/apps/calculator";

  @BeforeAll
  static void learn() {
    if (al.model().provider() == Provider.MISTRALAI) {
      al.learn(
          "4 / 2 =",
          List.of("click button '4'", "click button '÷'", "click button '2'", "click button '='"));
    }
  }

  @Test
  void testAddition() {
    navigate(CALCULATOR_URL);
    al.act("2 + 2 =");
    Assertions.assertEquals(4L, al.get("calculator result from textfield"));
  }

  @Test
  void testSubtraction() {
    navigate(CALCULATOR_URL);
    al.act("5 - 3 =");
    Assertions.assertEquals(2L, al.get("calculator result from textfield"));
  }

  @Test
  void testMultiplication() {
    navigate(CALCULATOR_URL);
    al.act("3 * 4 =");
    Assertions.assertEquals(12L, al.get("calculator result from textfield"));
  }

  @Test
  void testDivision() {
    navigate(CALCULATOR_URL);
    al.act("8 / 2 =");
    Assertions.assertEquals(4L, al.get("calculator result from textfield"));
  }
}
